axoproject/src/lib.rs

```diff
@@ -355,10 +355,16 @@ pub struct WorkspaceInfo {
     pub workspace_dir: Utf8PathBuf,
     /// Path to the root manifest of the workspace
     ///
-    /// This can be either a Cargo.toml or package.json. In either case this manifest
-    /// may or may not represent a "real" package. Both systems have some notion of
-    /// "virtual" manifest which exists only to list the actual packages in the workspace.
+    /// This can be either a Cargo.toml, package.json, dist.toml, or dist-workspace.toml.
+    ///
+    /// In most cases this manifest may or may not represent a "real" package. Many systems
+    /// have some notion of "virtual" manifest which exists only to list the actual packages
+    /// in the workspace.
     pub manifest_path: Utf8PathBuf,
+    /// Path to the dist.toml or dist-workspace.toml for the workspace
+    ///
+    /// If this is ever set, then this will be the same as manifest_path.
+    pub dist_manifest_path: Option<Utf8PathBuf>,
```

Comment:
Is it true that this is the same as `manifest_path` for a `package.json`/`Cargo.toml` workspace?

Reply:
The "if this is ever set" clause is doing heavy lifting here. Basically this option is morally a boolean saying manifest_path might be a dist(-workspace).toml, but I decided to make it an option to make code that deals with the pacakge version of dist_manifest_path uniform.

Reply:
Or i guess to put it another way -- if this is ever set, it's shared/generic workspace.